src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -1730,6 +1730,22 @@ bool checkValid(MethodSymbol candidate, bool isCandidate, DiagnosticBag specific
                         }
                     }
                 }
+                else if (LanguageVersion >= MessageID.IDS_FeatureAsyncMain.RequiredVersion() && taskEntryPoints.Count > 0 && viableEntryPoints.Count == 1)
+                {
+                    var taskCandidates = taskEntryPoints.Select(s => (Symbol)s.Candidate).ToImmutableArray();
```

Comment:
>Select [](start = 57, length = 6)

`SelectAsArray`

Reply:
@333fred Thanks for suggestion. Is this an extension method defined somewhere in Roslyn? I never met it before, and can't confirm as I'm not working in an IDE :-(

```suggestion
                    var taskCandidates = taskEntryPoints.SelectAsArray(s => (Symbol)s.Candidate);
```

Reply:
Yes, it's an extension we have on ImmutableArrays in roslyn.

---
In reply to: [473198509](https://github.com/dotnet/roslyn/pull/46832#discussion_r473198509) [](ancestors = 473198509)